From ea28c5b34457cf7676181b284e22ea5f79a30d85 Mon Sep 17 00:00:00 2001
From: "Maxin B. John" <maxin.john@intel.com>
Date: Thu, 13 Jul 2017 14:47:31 +0300
Subject: [PATCH] stdinc.h: fix build with mingw

Fixes this build error with mingw:
...
| compilation terminated.
| In file included from ../pkgconf-1.3.7/libpkgconf/libpkgconf.h:19:0,
| from ../pkgconf-1.3.7/libpkgconf/audit.c:16:
| ../pkgconf-1.3.7/libpkgconf/stdinc.h:36:12: fatal error: BaseTsd.h: No
such file or directory
| # include <BaseTsd.h>

Upstream-Status: Submitted

Signed-off-by: Maxin B. John <maxin.john@intel.com>
---
 libpkgconf/stdinc.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/libpkgconf/stdinc.h b/libpkgconf/stdinc.h
index ac7e53c..d8efcf5 100644
--- a/libpkgconf/stdinc.h
+++ b/libpkgconf/stdinc.h
@@ -33,10 +33,18 @@
 # include <malloc.h>
 # define PATH_DEV_NULL	"nul"
 # ifndef ssize_t
+# ifndef __MINGW32__
 #  include <BaseTsd.h>
+# else
+#  include <basetsd.h>
+# endif
 #  define ssize_t SSIZE_T
 # endif
-# include "win-dirent.h"
+# ifndef __MINGW32__
+#  include "win-dirent.h"
+# else
+# include <dirent.h>
+# endif
 #else
 # define PATH_DEV_NULL	"/dev/null"
 # include <dirent.h>
-- 
2.4.0

